### PR TITLE
Feat/search

### LINKED ISSUE #166: Feed: Search
We want to add a search functionality to the feed. this should use something like elastic search, debounce, etc. the backend might need to be updated for this

Image

output will look something like this with events to scroll through:

Image

### PR DESCRIPTION
**Pull request by AlenGanopolsky**
# Description
[Link to Ticket](https://github.com/GenerateNU/skillspark/issues/166)

Set up OpenSearch + Event index within AWS, created a supabase edge function to insert into OpenSearch on insert, update or delete of an event record, and created the OpenSearch go client with a corresponding fuzzy search method.

Now includes Terraform for the OpenSearch resource including events index that is being used within the OpenSearch instance

# How Has This Been Tested?
Manual testing using scalar

## Screenshots
Please provide screenshots of manual testing (Scalar, frontend pages, etc.)
image
image
image

# Checklist

## General
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas

## Backend Changes (if applicable)
- [ ] I have written unit tests for my code
- [ ] I have written integration tests where appropriate
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have run make api-validate and make api-gen to keep the openAPI spec up to date
- [ ] I have tested error handling and edge cases
- [ ] Database migrations are included (if applicable)

## Frontend Changes (if applicable)
- [ ] I have tested in multiple browsers (Chrome, Firefox, Safari)
- [ ] Mobile responsive design verified (if applicable)
- [ ] I have run make generate-api to keep the API hooks up to date
- [ ] **For mobile features:** Tested on both iOS and Android devices
- [ ] **For mobile features:** Screenshots from both iOS and Android included below
- [ ] No console errors or warnings
- [ ] Loading states and error states handled

### iOS
[Add iOS screenshots here]

### Android
[Add Android screenshots here]

### CODE REVIEW

**Comment by josh-torre** [comment]
@AlenGanopolsky is this ready to be reviewed?

**Comment by AlenGanopolsky** [comment]
> @AlenGanopolsky is this ready to be reviewed?

yes

File: frontend/apps/mobile/contexts/auth-context.tsx [hunk 1]
```diff
@@ -148,6 +148,7 @@ export function AuthProvider({ children }: { children: ReactNode }) {
       },
       {
         onSuccess: async (resp: signupGuardianResponse) => {
+          console.log('success!');
```
Comment (josh-torre) [inline frontend/apps/mobile/contexts/auth-context.tsx]: pls remove

File: frontend/apps/web/Dockerfile [hunk 2]
```diff
@@ -14,6 +14,7 @@ RUN bun install
 WORKDIR /app/apps/web
 
 RUN bun run build
+#RUN bun x tsc -b && bun x vite build
```
Comment (josh-torre) [inline frontend/apps/web/Dockerfile]: Pls remove the comment or uncomment the line

File: backend/internal/storage/postgres/storage.go [hunk 3]
```diff
@@ -16,6 +16,8 @@ import (
 func ConnectDatabase(ctx context.Context, dbConfig config.DB) (*pgxpool.Pool, error) {
 	log.Printf("Pool config - MaxConns: %d, MinConns: %d, MaxLifetime: %d",
 		dbConfig.MaxOpenConns, dbConfig.MaxIdleConns, dbConfig.ConnMaxLifetime)
+	log.Printf("Connecting to DB host=%s port=%s user=%s dbname=%s sslmode=%s",
```
Comment (adescoteaux1) [inline backend/internal/storage/postgres/storage.go]: remove

File: backend/internal/supabase/functions/opensearch-access/index.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
+import "jsr:@supabase/functions-js/edge-runtime.d.ts";
+
+const OPENSEARCH_URL = Deno.env.get("OPENSEARCH_URL")!;
+const OPENSEARCH_USER = Deno.env.get("OPENSEARCH_USER")!;
+const OPENSEARCH_PASS = Deno.env.get("OPENSEARCH_PASS")!;
+const INDEX = "events";
```
Comment (adescoteaux1) [inline backend/internal/supabase/functions/opensearch-access/index.ts:6]: do these need to be added to supabase settings?

File: backend/internal/supabase/functions/opensearch-access/index.ts
Comment (adescoteaux1) [inline backend/internal/supabase/functions/opensearch-access/index.ts:1]: has this been tested?

File: frontend/apps/mobile/contexts/auth-context.tsx (hunk 1 above)
Comment (adescoteaux1) [inline frontend/apps/mobile/contexts/auth-context.tsx]: remove

File: backend/internal/supabase/functions/opensearch-access/index.ts
Comment (AlenGanopolsky) [reply backend/internal/supabase/functions/opensearch-access/index.ts:1]: yes - but for good measure Ill make sure to test again

File: backend/internal/supabase/functions/opensearch-access/index.ts (hunk 4 above)
Comment (AlenGanopolsky) [reply backend/internal/supabase/functions/opensearch-access/index.ts:6]: good catch - yes they do. For any local instance, having these values set in backend/internal/supabase/functions is enough

File: backend/internal/supabase/seed/event_edge_function.sql [hunk 5]
```diff
@@ -0,0 +1,9 @@
+-- NOTE - run each chunk of sql code depending on what lane is being deployed (uncomment/comment and then run supabase db reset)
+
+-- dev
+ALTER DATABASE postgres SET app.settings.edge_function_url = 'http://supabase_edge_runtime_skillspark:8081/opensearch-access';
+ALTER DATABASE postgres SET app.settings.edge_function_key = 'Bearer sb_publishable_ACJWlzQHlZjBrEguHvfOxg_3BJgxAaH';
```
Comment (adescoteaux1) [inline backend/internal/supabase/seed/event_edge_function.sql]: is this the real env value?

Comment (AlenGanopolsky) [reply backend/internal/supabase/seed/event_edge_function.sql]: Yes, I pulled the key from supabase in prod. I can double check again, but I just went to https://supabase.com/dashboard/project/jpxypbmcgjhirlaatufp/settings/api-keys and pulled it from there

**Comment by josh-torre** [comment]
@AlenGanopolsky Please fix tests

**Review by josh-torre — COMMENTED** [review]
Pending @adescoteaux1 review

File: frontend/apps/web/Dockerfile [hunk 6]
```diff
@@ -13,7 +13,8 @@ RUN bun install
 # Build the web app
 WORKDIR /app/apps/web
 
-RUN bun run build
+#RUN bun run build
+RUN bun x tsc -b && bun x vite build
```
Comment (adescoteaux1) [inline frontend/apps/web/Dockerfile]: undo this

**Comment by AlenGanopolsky** [comment]
added some additional stuff to the trigger to handle errors gracefully - which fixed the issue which made all tests fail due to poor error handling

**Review by adescoteaux1 — APPROVED** [review]
